fe/fe-core/src/main/java/org/apache/doris/persist/EditLog.java

```diff
@@ -1745,6 +1777,50 @@ public void logColocateModifyRepliaAlloc(ColocatePersistInfo info) {
         logEdit(OperationType.OP_COLOCATE_MOD_REPLICA_ALLOC, info);
     }
 
+    public void logRestoreAndColocateAddTable(RestoreJob job, List<ColocatePersistInfo> infos) {
+        long start = System.currentTimeMillis();
+
+        List<Pair<Short, Writable>> list = new ArrayList<>();
+        list.add(Pair.of(OperationType.OP_RESTORE_JOB, job));
+
+        for (ColocatePersistInfo info : infos) {
+            list.add(Pair.of(OperationType.OP_COLOCATE_ADD_TABLE, info));
+        }
+
+        try {
+            logEdit(list);
+        } catch (Exception e) {
+            LOG.warn("failed to log ColocateAddTable", e);
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("log ColocateAddTable {} . cost: {} ms", infos.size(),
+                    (System.currentTimeMillis() - start));
+        }
+    }
```

Comment:
And then instance a `JournalBatch` here instead of `List<Pair<Short, Writeable>>`.